lib/strong_migrations/migration.rb

```diff
@@ -186,6 +195,24 @@ def backfill_code(table, column, default)
       end
     end
 
+    def foreign_key_code(from_table, to_table)
+      if ActiveRecord::VERSION::MAJOR > 5 || ActiveRecord::VERSION::MAJOR == 5 && ActiveRecord::VERSION::MINOR >= 2
+        <<~CODE.strip
+          #{command_str(:add_foreign_key, [from_table, to_table, validate: false])}
+              #{command_str(:validate_foreign_key, [from_table, to_table])}
```

Comment:
I'm not certain that it's safe to validate the constraint in the same transaction as the foreign key being added. I'm still learning about these aspects of database management. Do you have any insight?

Reply:
I'm pretty sure they need to be in separate transactions.